src/meltano/cli/install.py

```diff
@@ -110,9 +114,19 @@ def install(  # noqa: C901
 def _get_schedule_plugins(project: Project, schedule_name: str):
     schedule_service = ScheduleService(project)
     schedule_obj = schedule_service.find_schedule(schedule_name)
-    task_sets = schedule_service.task_sets_service.get(schedule_obj.job)
-    schedule_plugins = []
-    for plugin_command in task_sets.flat_args:
-        plugin_name = plugin_command.split(":")[0]
-        schedule_plugins.append(project.plugins.find_plugin(plugin_name))
+    schedule_plugins = set()
+    if schedule_obj.elt_schedule:
+        for plugin_name in schedule_obj.elt_args[:2]:
+            schedule_plugins.add(project.plugins.find_plugin(plugin_name))
+    else:
+        task_sets = schedule_service.task_sets_service.get(schedule_obj.job)
+        for blocks in task_sets.flat_args_per_set:
+            parser = BlockParser(logger, project, blocks)
+            for plugin in parser.plugins:
+                if plugin.type == PluginType.MAPPERS:
+                    schedule_plugins.add(
+                        project.plugins.find_plugin(plugin.info.get("name")),
+                    )
+                else:
+                    schedule_plugins.add(plugin)
```

Comment:
@WillDaSilva thanks for the suggestion. Good point, I agree this logic could be more concise. What do you think about this more concise if/then logic instead of ternary though? I personally find ternaries harder to read.

```python
for plugin in parser.plugins:
    if plugin.type == PluginType.MAPPERS:
        plugin = project.plugins.find_plugin(plugin.info.get("name"))
    schedule_plugins.add(plugin)
```